lacework/resource_lacework_vulnerability_exception_host.go

```diff
@@ -0,0 +1,421 @@
+	return nil
+}
+
+func resourceLaceworkVulnerabilityExceptionHostUpdate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework   = meta.(*api.Client)
+		severities = castAttributeToStringSlice(d, "vulnerability_criteria.0.severities")
+		packages   = castAttributeToArrayOfCustomKeyValueMap(d, "vulnerability_criteria.0.package", "name", "version")
+	)
+
+	// we validate the time format in the schema validate func, so we can ignore the error here
+	expiry, _ := time.Parse(time.RFC3339, d.Get("expiration_time").(string))
+
+	vulnException := api.NewVulnerabilityException(d.Get("name").(string),
+		api.VulnerabilityExceptionConfig{
+			Description:     d.Get("description").(string),
+			Type:            api.VulnerabilityExceptionTypeHost,
+			ExceptionReason: api.NewVulnerabilityExceptionReason(d.Get("reason").(string)),
+			Severities:      api.NewVulnerabilityExceptionSeverities(severities),
+			Cve:             castAttributeToStringSlice(d, "vulnerability_criteria.0.cves"),
+			Package:         api.NewVulnerabilityExceptionPackages(packages),
+			Fixable:         d.Get("vulnerability_criteria.0.fixable").(bool),
+			ResourceScope: api.VulnerabilityExceptionHostResourceScope{
+				Hostname:    castAttributeToStringSlice(d, "resource_scope.0.hostnames"),
+				ExternalIP:  castAttributeToStringSlice(d, "resource_scope.0.external_ips"),
+				ClusterName: castAttributeToStringSlice(d, "resource_scope.0.cluster_names"),
+				Namespace:   castAttributeToStringSlice(d, "resource_scope.0.namespaces"),
+			},
+			ExpiryTime: expiry.UTC(),
+		},
+	)
+
+	if !d.Get("enabled").(bool) {
+		vulnException.Enabled = 0
+	}
+
+	log.Printf("[INFO] Updating vulnerability exception with data:\n%+v\n", vulnException)
+	response, err := lacework.V2.VulnerabilityExceptions.UpdateVulnerabilityExceptionsHost(vulnException, d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("name", response.Data.ExceptionName)
+	d.Set("guid", response.Data.Guid)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_time", response.Data.CreatedTime)
+	d.Set("updated_time", response.Data.UpdatedTime)
+	d.Set("created_by", response.Data.Props.CreatedBy)
+	d.Set("updated_by", response.Data.Props.UpdatedBy)
+	d.Set("type", response.Data.ExceptionType)
+
+	log.Printf("[INFO] Updated vulnerability exception with guid %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkVulnerabilityExceptionHostDelete(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Deleting vulnerability exception with guid %s\n", d.Id())
+	err := lacework.V2.VulnerabilityExceptions.Delete(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Deleted vulnerability exception with guid %s\n", d.Id())
+	return nil
+}
+
+func vulnerabilityExceptionReasonsSlice()[]string{
+	var reasons []string
+	for  _, value := range api.VulnerabilityExceptionReasons {
```

Comment:
This is marking an fmt error. I wonder why we are not running this on our pipeline?!!!

Reply:
Pipeline is complaining but Github timed out to report the status https://g.codefresh.io/build/61ddb4b6090d7c7bed502fae

So weird!